llvm/lib/IR/StructuralHash.cpp

```diff
@@ -98,14 +123,21 @@ class StructuralHashImpl {
       return stable_hash_combine(Hashes);
     }
 
+    if (auto *GVar = dyn_cast<GlobalVariable>(C)) {
+      Hashes.emplace_back(hashGlobalVariable(*GVar));
+      return stable_hash_combine(Hashes);
+    }
+
     if (auto *G = dyn_cast<GlobalValue>(C)) {
       Hashes.emplace_back(hashGlobalValue(G));
       return stable_hash_combine(Hashes);
     }
 
     if (const auto *Seq = dyn_cast<ConstantDataSequential>(C)) {
-      Hashes.emplace_back(xxh3_64bits(Seq->getRawDataValues()));
-      return stable_hash_combine(Hashes);
+      if (Seq->isString()) {
```

Comment:
What about other `ConstantDataSequential`s which are not string?

Reply:
If you see below in the switch statement, `ConstantArrayVal` or `ConstantVectorVal` (that inherits `ConstantDataSequentials`) should cover the rest by handling each Constant element.